contracts/contracts/lib/AssetHelper.sol

```diff
@@ -192,15 +197,29 @@ library SupplySourceHelper {
     }
 
     /// @notice Gets the balance in our treasury.
-    function balance(SupplySource memory supplySource) internal view returns (uint256 ret) {
-        if (supplySource.kind == SupplyKind.Native) {
+    function balance(Asset memory asset) internal view returns (uint256 ret) {
+        if (asset.kind == AssetKind.Native) {
             ret = address(this).balance;
-        } else if (supplySource.kind == SupplyKind.ERC20) {
-            ret = IERC20(supplySource.tokenAddress).balanceOf(address(this));
+        } else if (asset.kind == AssetKind.ERC20) {
+            ret = IERC20(asset.tokenAddress).balanceOf(address(this));
         }
     }
 
-    function native() internal pure returns (SupplySource memory) {
-        return SupplySource({kind: SupplyKind.Native, tokenAddress: address(0)});
+    function native() internal pure returns (Asset memory) {
+        return Asset({kind: AssetKind.Native, tokenAddress: address(0)});
+    }
+
+    function isNative(Asset memory self) internal pure returns(bool) {
+        return self.kind == AssetKind.Native;
+    }
+
+    function increaseAllowance(Asset memory self, address spender, uint256 amount) internal {
+        if (self.kind == AssetKind.ERC20) {
+            IERC20 token = IERC20(self.tokenAddress);
+            uint256 allowance = token.allowance(address(this), spender);
+            if (!token.approve(spender, allowance + amount)) {
+                revert IncreaseAllowanceFailed();
+            }
+        }
```

Comment:
1. Just use `SafeERC20#safeIncreaseAllowance()`?
2. Singleton ifs like this are cleaner if written as a negative if with an early return.